app/assets/javascripts/index/history.js

```diff
@@ -173,7 +178,14 @@ OSM.History = function (map) {
 
     map.on("zoomend", updateBounds);
 
-    loadFirstChangesets();
+    var maxId;
+    var paramsIndex = path.indexOf("?");
+    if (paramsIndex >= 0) {
+      var params = Qs.parse(path.substring(paramsIndex + 1));
+      maxId = params.max_id;
+    }
+
+    loadFirstChangesets(maxId);
```

Comment:
I don't understand the point of this, indeed of any of this commit, when you remove it all again in the next commit?